nodes/viz/texture_viewer.py

```diff
@@ -0,0 +1,286 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+
+from mathutils import Vector
+import bpy
+from bpy.props import FloatProperty, EnumProperty, StringProperty, BoolProperty
+
+import blf
+import bgl
+
+from sverchok.data_structure import updateNode, node_id
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.ui import nodeview_bgl_viewer_draw_mk2 as nvBGL2
+
+palette_dict = {
+    "default": (
+        (0.243299, 0.590403, 0.836084, 1.00),  # back_color
+        (0.390805, 0.754022, 1.000000, 1.00),  # grid_color
+        (1.000000, 0.330010, 0.107140, 1.00)   # line_color
+    ),
+    "scope": (
+        (0.274677, 0.366253, 0.386430, 1.00),  # back_color
+        (0.423268, 0.558340, 0.584078, 1.00),  # grid_color
+        (0.304762, 1.000000, 0.062827, 1.00)   # line_color
+    )
+
+}
+
+size_tex_list=[
+    ('EXTRA_SMALL','extra_small 64x64px','extra small squared tex: 64px','',64),
+    ('SMALL','small 128x128px','small squared tex: 128px','',128),
+    ('MEDIUM','medium 256x256px','medium squared tex: 256px','',256),
+    ('LARGE','large 512x512px','large squared tex: 512px','',512),
+    ('EXTRA_LARGE','extra_large 1024x1024px','extra large squared tex: 1024px','',1024)
+]
+
+size_tex_dict = {
+    'EXTRA_SMALL': 64,
+    'SMALL': 128,
+    'MEDIUM': 256,
+    'LARGE': 512,
+    'EXTRA_LARGE': 1024
+}
+
+bitmap_save_list=[
+    ('PNG','png format', 'save texture in .png fromat','',0),
+    ('TGA','tga format','save texture in .tga fromat','',1),
+    ('TIFF','tiff format','save texture in .tiff format','',2),
+    ('BMP','bmp format','save texture in .tiff format','',3),
+    ('JPEG','jpeg format','save texture in .jpeg format','',4)
+]
+def assign_image(self,image_name, size ,buffer):
```

Comment:
Here remove self and pass the image reference instead of name since you create the image somewhere else.